keygrabber.c

```diff
@@ -31,6 +31,108 @@
 
 #include "keygrabber.h"
 #include "globalconf.h"
+#include "common/luaclass.h"
+
+static int miss_index_handler    = LUA_REFNIL;
+static int miss_newindex_handler = LUA_REFNIL;
+static int miss_call_handler     = LUA_REFNIL;
+
+/**
+* Allow the Lua handler to be implemented for custom properties and
+* functions.
+* \param L A lua state
+* \param handler A function on the LUA_REGISTRYINDEX
+*/
+static int
+luaA_keygrabber_call_handler(lua_State *L, int handler)
+{
+    int nargs = lua_gettop(L);
+
+    /* Push error handling function and move it before args */
+    lua_pushcfunction(L, luaA_dofunction_error);
+    lua_insert(L, - nargs - 1);
+    int error_func_pos = 1;
+
+    /* Push function and move it before args */
+    lua_rawgeti(L, LUA_REGISTRYINDEX, handler);
+    lua_insert(L, - nargs - 1);
+
+    if(lua_pcall(L, nargs, LUA_MULTRET, error_func_pos))
+    {
+        warn("%s", lua_tostring(L, -1));
+        /* Remove error function and error string */
+        lua_pop(L, 2);
+        return 0;
+    }
+    /* Remove error function */
+    lua_remove(L, error_func_pos);
+
+    return lua_gettop(L);
+}
+
+/**
+* Add a custom property handler (getter).
+*/
+static int
+luaA_keygrabber_set_call_handler(lua_State *L)
+{
+    return luaA_registerfct(L, 1, &miss_call_handler);
+}
+
+/**
+* Add a custom property handler (getter).
+*/
+static int
+luaA_keygrabber_set_index_miss_handler(lua_State *L)
```

Comment:
not a setter?

Reply:
This is a getter metatable handler, lua call the setter `newindex`

Reply:
so should it be `luaA_keygrabber_get_index_miss_handler`?

Reply:
no, it *sets* the *getter* metatable handler

This is the C equivalent of `setmetatable({}, __index = function() end})`